Sources/Sentry/SentryClient.m

```diff
@@ -641,7 +641,13 @@ - (void)setSdk:(SentryEvent *)event
     id integrations = event.extra[@"__sentry_sdk_integrations"];
     if (!integrations) {
         integrations = [NSMutableArray new];
-        for (NSString *integration in SentrySDK.currentHub.installedIntegrationNames) {
+
+        // The SDK can send an event before it installs all integrations, which add to this list. To
+        // avoid an exception while iterating over a list while items get added to it, we copy it.
+        // With that approach, we might not have the complete integration list. To fix this, the SDK
+        // would need to ask all integrations first if they should be installed before actually
+        // installing them, which would be quite a bit a of a refactoring.
```

Comment:
I don't think it's a big refactor though. All the code is contained in this one function: https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentrySDK.m#L368-L396.

We'd just loop twice: first we run the `shouldBeEnabledWithOptions` method which all integrations already have. If yes, we add the integration name to `installedIntegrationNames` (maybe rename to `enabledIntegrationNames`).

Then we run the loop again, now actually installing them and adding it to `installedIntegrations`.

So I think "quite a bit a of a refactoring" is in fact really a small one if you want to prevent the list from not being correct.

Reply:
The problem is that `shouldBeEnabledWithOptions` is not part of `SentryIntegrationProtocol`, which is public. If someone would add their custom integration, this wouldn't work correctly. Anyways, we need to make the integrations on the hub thread-safe, as you can call `captureEvent` shortly after initializing the SDK, or in your own integration, or you could also call `SentrySDK.close` on another thread. I'm not going to use `shouldBeEnabledWithOptions` for now.

Reply:
Good point!